Add post tags

onuraltinkurt opened the pull request.

Commit:
Allow users to choose predefined tags during post creation

Commit:
Merge branch 'dev' of github.com:SWE-574-Group-4/Community-Management-System into addPostTags

Commit:
migration file

mehmeteyupoglu commented on .DS_Store:
This should not be pushed as it doesn't have to do anything with the features requested

onuraltinkurt replied:
Thanks. I took it back.

mehmeteyupoglu commented on client/src/views/post/components/MapFields.tsx:
```diff
@@ -99,6 +101,30 @@ export default function MapFields({ fields }: { fields: FieldType[] }) {
+    useEffect(() => {
+        const fetchTags = async () => {
+            try {
+                const response = await axios.get('http://localhost:8000/tags/')
```
We have a service and many examples of its use in the system. Let's stick with it. Example: `apiGetCommunityTemplates`

onuraltinkurt replied:
Can you guide me with this?

mehmeteyupoglu commented on client/src/views/post/components/MapFields.tsx:
```diff
@@ -108,10 +134,14 @@ export default function MapFields({ fields }: { fields: FieldType[] }) {
         // turn form data into a string with JSON.stringify
         if (typeof handlePost === 'function') {
-            handlePost(id, userId, JSON.stringify(formData))
+            handlePost(id, userId, formData,tags)
```
Also we need to discuss why we don't need `JSON.stringify(formData)` anymore. It's an important part of the code

onuraltinkurt replied:
You can have a look at PostService.ts
I select the header type as json there. then we do not need to convert formdata to json in mapfields.tsx anymore. 
{
    return ApiService.fetchData({
        url: `/post/`,
        method: 'post',
        headers: {
            'Content-Type': 'application/json'
          },
        data: {
            community_id: communityId,
            user_id: userId,
            content: fields,
            tag_ids:tag_ids
        },
    })
}

mehmeteyupoglu reviewed (changes requested):
Rashid also works on tags and labels. You both will have conflicts since you are using the same fields and table names. You need to come together before having merge conflicts or migrations conflicts.

Commit:
remove .DS_Store

Commit:
Merge branch 'dev' of github.com:SWE-574-Group-4/Community-Management-System into addPostTags

mehmeteyupoglu reviewed (changes requested):
When I try to reproduce the functionality, I bumped into these issues

- no tags are visible (probably due to the seeding, you may need to seed data)
- Also the select component should be used from this [[link](https://elstar.themenate.net/ui-components/select)](url)
- Also there were conflicts and I removed them

mehmeteyupoglu committed:
Add badge and notification features, update routes, and modify settings

- Introduced Badge and Notification types in TypeScript
- Added API services for fetching user badges and notifications
- Updated routes for report creation and viewing
- Modified settings to include MEDIA_URL and MEDIA_ROOT
- Added Vercel configuration for rewrites
- Updated requirements.txt to include new dependencies
- Created migration files for database schema changes

mehmeteyupoglu committed:
Refactor PostService to include apiGetTags function and update MapFields to use it for fetching tags

mehmeteyupoglu committed:
Refactor MapFields component to use custom Select component and update imports

mehmeteyupoglu committed:
Add migration to create default tags in the Tag model

mehmeteyupoglu committed:
Prevent duplicate tag creation in migration

mehmeteyupoglu committed:
Add conditional rendering for post title in DisplayPost component

mehmeteyupoglu committed:
Merge branch 'dev' into addPostTags

mehmeteyupoglu committed:
Refactor apiPost function to improve header management and add TODO comments for future enhancements

onuraltinkurt committed:
Merge branch 'dev' of github.com:SWE-574-Group-4/Community-Management-System into addPostTags

onuraltinkurt committed:
post tags issue fix

onuraltinkurt committed:
addPost fix

onuraltinkurt committed:
fix conflicts
